service/collector/vmss_rate_limit.go

```diff
@@ -114,124 +120,164 @@ func (u *VMSSRateLimit) Collect(ch chan<- prometheus.Metric) error {
 	}
 	autorest.StatusCodesForRetry = codes
 
-	// We need all CRs to gather all subscriptions below.
-	var crs []providerv1alpha1.AzureConfig
-	{
-		mark := ""
-		page := 0
-		for page == 0 || len(mark) > 0 {
-			opts := metav1.ListOptions{
-				Continue: mark,
-			}
-			list, err := u.g8sClient.ProviderV1alpha1().AzureConfigs(metav1.NamespaceAll).List(ctx, opts)
-			if err != nil {
-				return microerror.Mask(err)
-			}
-
-			crs = append(crs, list.Items...)
-
-			mark = list.Continue
-			page++
-		}
+	clustersSecret, err := u.getClusters(ctx)
+	if err != nil {
+		return microerror.Mask(err)
 	}
 
-	{
-		var doneSubscriptions []string
-		for _, cr := range crs {
-			config, err := credential.GetAzureConfigFromSecretName(ctx, u.k8sClient, key.CredentialName(cr), key.CredentialNamespace(cr), u.gsTenantID)
-			if err != nil {
-				return microerror.Mask(err)
-			}
+	var doneSubscriptions []string
+	for cluster, secret := range clustersSecret {
+		config, err := credential.GetAzureConfigFromSecret(secret, u.gsTenantID)
+		if err != nil {
+			return microerror.Mask(err)
+		}
 
-			// We want to check only once per subscriptino
-			if inArray(doneSubscriptions, config.SubscriptionID) {
-				continue
-			}
+		// We want to check only once per subscription
+		if inArray(doneSubscriptions, config.SubscriptionID) {
+			continue
+		}
 
-			azureClients, err := client.NewAzureClientSet(*config)
-			if err != nil {
-				return microerror.Mask(err)
-			}
+		azureClients, err := client.NewAzureClientSet(*config)
+		if err != nil {
+			return microerror.Mask(err)
+		}
 
-			// VMSS List VMs specific limits.
-			{
-				var headers []string
+		result, err := azureClients.VirtualMachineScaleSetVMsClient.ListComplete(ctx, cluster, fmt.Sprintf("%s-master-%s", cluster, cluster), "", "", "")
+		if IsThrottlingError(err) {
```

Comment:
We can only expose `measuredCalls` metrics when hitting 429, so I made it explicit that we are checking that the error is a throttling error.